internal/resolve/catalog.go

```diff
@@ -190,6 +178,42 @@ func (r *CatalogResolver) Resolve(ctx context.Context, ext *ocv1alpha1.ClusterEx
 	return resolvedBundle, resolvedBundleVersion, priorDeprecation, nil
 }
 
+type resolutionErrorInfo struct {
```

Comment:
Should we just make this type implement `error` and return it directly. (And then change `func buildResolutionError` to be `func (rei resolutionErrorInfo) Error() string`?

Reply:
That sounds reasonable to me